[MIHOME]  "zero-fire" 1 Channel Wall Switch not working

I have 2  Xiaomi "zero-fire" Wall Switch
1 is one button and another one is 2 buttons.

2 buttons works fine and can be auto discovered but 1 button wall switch cant be discovered.
Manually add the thing and show online but actually not working.

**openhab.log :** 

[ERROR] [ndler.XiaomiAqaraActorSwitch1Handler] - Can't handle command ON on channel mihome:ctrl_ln1:158d0001f99045:ch1


**MI Home APPS gateway information :** 

version code：158
网关ID57831448
Zigbee通道：0
网关信息：
null

子设备信息：
**[{"model":"lumi.ctrl_ln1.aq1","did":"lumi.158d0001f99045","name":"AC3"},**  **------> not working**
{"model":"lumi.ctrl_ln2.v1","did":"lumi.158d0001da64da","name":"AC2 \/ AC1"},{"model":"lumi.sensor_ht.v1","did":"lumi.158d0001c2c368","name":"A1"},{"model":"lumi.sensor_ht.v1","did":"lumi.158d0001c2aeea","name":"A2"},{"model":"lumi.sensor_ht.v1","did":"lumi.158d0001c2b7ea","name":"A3"},{"model":"lumi.sensor_ht.v1","did":"lumi.158d0001b70b02","name":"MIS Temperature"},{"model":"lumi.sensor_ht.v1","did":"lumi.158d0001b70b32","name":"Server Room Temperature"},{"model":"lumi.sensor_ht.v1","did":"lumi.158d0001c2af3c","name":"Voice Temperature"}]

I'm using OH2 snapshot and uploaded the latest mihome binding from 
https://openhab.jfrog.io/openhab/libs-pullrequest-local/org/openhab/binding/org.openhab.binding.mihome/2.2.0-SNAPSHOT/




Thanks.

Hi, Anyone can help?

Debug Log : 


**15:24:35.456 [DEBUG] [ng.mihome.handler.XiaomiBridgeHandler] - Adding item listener for device 158d0001f99045**
15:24:35.460 [DEBUG] [ng.mihome.handler.XiaomiBridgeHandler] - Having 9 Item listeners
**15:24:35.463 [DEBUG] [ihome.handler.XiaomiDeviceBaseHandler] - Device 158d0001f99045 on channel mihome:ctrl_ln1:158d0001f99045:ch1 received command REFRESH**
**15:24:35.469 [DEBUG] [ihome.handler.XiaomiDeviceBaseHandler] - Update Item 158d0001f99045 with retented message**
**15:24:35.472 [DEBUG] [ihome.handler.XiaomiDeviceBaseHandler] - Item got update: {"cmd":"read_ack","model":"","sid":"158d0001f99045","short_id":21824,"data":"{\"voltage\":3600,\"channel_0\":\"off\"}"}**
15:24:39.724 [DEBUG] [g.mihome.internal.socket.XiaomiSocket] - Received Datagram from 192.168.128.132:4321 on Port 9898
15:24:39.730 [DEBUG] [ihome.handler.XiaomiDeviceBaseHandler] - Item got update: {"cmd":"report","model":"sensor_ht","sid":"158d0001b70b02","short_id":42769,"data":"{\"temperature\":\"2036\"}"}
15:24:39.737 [DEBUG] [g.mihome.internal.socket.XiaomiSocket] - Received Datagram from 192.168.128.132:4321 on Port 9898
15:24:39.742 [DEBUG] [ihome.handler.XiaomiDeviceBaseHandler] - Item got update: {"cmd":"report","model":"sensor_ht","sid":"158d0001b70b02","short_id":42769,"data":"{\"humidity\":\"4340\"}"}
**15:24:41.002 [DEBUG] [ihome.handler.XiaomiDeviceBaseHandler] - Device 158d0001f99045 on channel mihome:ctrl_ln1:158d0001f99045:ch1 received command REFRESH**
**15:24:44.592 [DEBUG] [ihome.handler.XiaomiDeviceBaseHandler] - Device 158d0001f99045 on channel mihome:ctrl_ln1:158d0001f99045:ch1 received command ON**
**15:24:44.605 [DEBUG] [ng.mihome.handler.XiaomiBridgeHandler] - Send to bridge mihome:bridge:34ce00907c65: {"cmd": "write", "sid": "158d0001f99045", "data": "{\"channel_0\": \"on\", \"key\": \"AED9B93938D009E135F6A92BC4052939"}"}
15:24:44.616 [ERROR] [andler.XiaomiAqaraActorSwitch1Handler] - Can't handle command ON on channel mihome:ctrl_ln1:158d0001f99045:ch1**
15:24:44.625 [DEBUG] [g.mihome.internal.socket.XiaomiSocket] - Received Datagram from 192.168.128.132:9898 on Port 9898
**15:24:44.630 [DEBUG] [ng.mihome.handler.XiaomiBridgeHandler] - Device 158d0001f99045 honored write request
15:24:44.634 [DEBUG] [ihome.handler.XiaomiDeviceBaseHandler] - Item got update: {"cmd":"write_ack","sid":"158d0001f99045","data":"{\"error\":\"Unsupported device\"}"}**
**15:24:45.194 [DEBUG] [ihome.handler.XiaomiDeviceBaseHandler] - Device 158d0001f99045 on channel mihome:ctrl_ln1:158d0001f99045:ch1 received command OFF
15:24:45.198 [DEBUG] [ng.mihome.handler.XiaomiBridgeHandler] - Send to bridge mihome:bridge:34ce00907c65: {"cmd": "write", "sid": "158d0001f99045", "data": "{\"channel_0\": \"off\", \"key\": \"AED9B93938D009E135F6A92BC4052939"}"}
15:24:45.203 [ERROR] [andler.XiaomiAqaraActorSwitch1Handler] - Can't handle command OFF on channel mihome:ctrl_ln1:158d0001f99045:ch1**
15:24:45.217 [DEBUG] [g.mihome.internal.socket.XiaomiSocket] - Received Datagram from 192.168.128.132:9898 on Port 9898
**15:24:45.219 [DEBUG] [ng.mihome.handler.XiaomiBridgeHandler] - Device 158d0001f99045 honored write request
15:24:45.221 [DEBUG] [ihome.handler.XiaomiDeviceBaseHandler] - Item got update: {"cmd":"write_ack","sid":"158d0001f99045","data":"{\"error\":\"Unsupported device\"}"}**

--- a/addons/binding/org.openhab.binding.mihome/ESH-INF/binding/binding.xml
+++ b/addons/binding/org.openhab.binding.mihome/ESH-INF/binding/binding.xml
@@ -1,8 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
-<binding:binding id="mihome"
-		xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-		xmlns:binding="http://eclipse.org/smarthome/schemas/binding/v1.0.0"
-		xsi:schemaLocation="http://eclipse.org/smarthome/schemas/binding/v1.0.0 http://eclipse.org/smarthome/schemas/binding-1.0.0.xsd">
+<binding:binding id="mihome" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:binding="http://eclipse.org/smarthome/schemas/binding/v1.0.0"
+	xsi:schemaLocation="http://eclipse.org/smarthome/schemas/binding/v1.0.0 http://eclipse.org/smarthome/schemas/binding-1.0.0.xsd">
 
 	<name>Xiaomi Mi Smart Home Binding</name>
 	<description>Connects to Xiaomi Smart Gateway v2 and allows to communicate with Xiaomi Mi Smart Home suite (also called Xiaomi Aqara). 

--- a/addons/binding/org.openhab.binding.mihome/ESH-INF/thing/aqara_86sw1.xml
+++ b/addons/binding/org.openhab.binding.mihome/ESH-INF/thing/aqara_86sw1.xml
@@ -19,5 +19,5 @@
 		</channels>
 		<config-description-ref uri="thing-type:mihome:device" />
 	</thing-type>
-	
+
 </thing:thing-descriptions>

--- a/addons/binding/org.openhab.binding.mihome/ESH-INF/thing/aqara_ctrl_ln1.xml
+++ b/addons/binding/org.openhab.binding.mihome/ESH-INF/thing/aqara_ctrl_ln1.xml
@@ -15,5 +15,5 @@
 		</channels>
 		<config-description-ref uri="thing-type:mihome:device" />
 	</thing-type>
-	
+
 </thing:thing-descriptions>

--- a/addons/binding/org.openhab.binding.mihome/ESH-INF/thing/channel.xml
+++ b/addons/binding/org.openhab.binding.mihome/ESH-INF/thing/channel.xml
@@ -18,7 +18,8 @@
 	<channel-type id="brightness">
 		<item-type>Dimmer</item-type>
 		<label>Brightness</label>
-		<description>The brightness channel allows to control the brightness of a light. It is also possible to switch the light on and off.</description>
+		<description>The brightness channel allows to control the brightness of a light. It is also possible to switch the
+			light on and off.</description>
 		<category>DimmableLight</category>
 		<tags>
 			<tag>Lighting</tag>
@@ -66,6 +67,11 @@
 		<description>This channel controls the volume of the gateway for playing sounds</description>
 		<category>SoundVolume</category>
 		<state max="100" min="0" step="1" pattern="%d %%" readOnly="false"></state>
+	</channel-type>
+
+	<channel-type id="joinPermission">
+		<item-type>Switch</item-type>
+		<label>Enable Device Inclusion</label>
 	</channel-type>
 
 	<channel-type id="enableSound">

--- a/addons/binding/org.openhab.binding.mihome/ESH-INF/thing/curtain.xml
+++ b/addons/binding/org.openhab.binding.mihome/ESH-INF/thing/curtain.xml
@@ -8,7 +8,8 @@
 			<bridge-type-ref id="bridge" />
 		</supported-bridge-type-refs>
 		<label>Xiaomi Aqara Intelligent Curtain Motor</label>
-		<description>Smart curtain controller for drawing back and forth fabric curtains. It has manual and remote operation modes.</description>
+		<description>Smart curtain controller for drawing back and forth fabric curtains. It has manual and remote operation
+			modes.</description>
 		<channels>
 			<channel id="curtainControl" typeId="curtainControl" />
 		</channels>

--- a/addons/binding/org.openhab.binding.mihome/ESH-INF/thing/gateway.xml
+++ b/addons/binding/org.openhab.binding.mihome/ESH-INF/thing/gateway.xml
@@ -10,6 +10,7 @@
 		<label>Xiaomi Mi Smart Home Gateway</label>
 		<description>Multifunctional Gateway - has a built in light and ambient light sensor and can play sounds</description>
 		<channels>
+			<channel typeId="joinPermission" id="joinPermission" />
 			<channel id="brightness" typeId="brightness" />
 			<channel id="illumination" typeId="illumination" />
 			<channel id="color" typeId="color" />

--- a/addons/binding/org.openhab.binding.mihome/ESH-INF/thing/sensor_cube.xml
+++ b/addons/binding/org.openhab.binding.mihome/ESH-INF/thing/sensor_cube.xml
@@ -8,7 +8,8 @@
 			<bridge-type-ref id="bridge" />
 		</supported-bridge-type-refs>
 		<label>Xiaomi Mi Smart Cube</label>
-		<description>Multifunctional controller equipped with an accelerometer and a gyroscope. Triggers the following actions: move, rotate, rotate, flip 90, flip 180, tap twice, shake air, free fall, alert.</description>
+		<description>Multifunctional controller equipped with an accelerometer and a gyroscope. Triggers the following
+			actions: move, rotate, rotate, flip 90, flip 180, tap twice, shake air, free fall, alert.</description>
 		<channels>
 			<channel id="action" typeId="action"></channel>
 			<channel id="rotationAngle" typeId="rotationAngle"></channel>

--- a/addons/binding/org.openhab.binding.mihome/ESH-INF/thing/sensor_ht.xml
+++ b/addons/binding/org.openhab.binding.mihome/ESH-INF/thing/sensor_ht.xml
@@ -9,7 +9,8 @@
 		</supported-bridge-type-refs>
 		<label>Xiaomi Mi Temperature &amp; Humidity Sensor</label>
 		<description>Reports temperature and humidity. Operating temperature: −20°C to 60°C. Operating humidity: 0 to 100%.
-		Sensor reports the temperature when there's a difference around 0.5°C. If there is no significant temperature change, sensor reports temperature once a 50 minutes.\
+			Sensor reports the temperature when there's a difference around 0.5°C. If there is no significant temperature change,
+			sensor reports temperature once a 50 minutes.\
 		</description>
 		<channels>
 			<channel id="temperature" typeId="temperature" />

--- a/addons/binding/org.openhab.binding.mihome/ESH-INF/thing/sensor_motion.xml
+++ b/addons/binding/org.openhab.binding.mihome/ESH-INF/thing/sensor_motion.xml
@@ -8,7 +8,8 @@
 			<bridge-type-ref id="bridge" />
 		</supported-bridge-type-refs>
 		<label>Xiaomi Mi Motion Sensor</label>
-		<description>Sensor that detects movement. Also called Occupancy Sensor or Human Body Sensor. After it detects motion, it goes to sleep for 1 minute.</description>
+		<description>Sensor that detects movement. Also called Occupancy Sensor or Human Body Sensor. After it detects motion,
+			it goes to sleep for 1 minute.</description>
 		<channels>
 			<channel id="motion" typeId="motion" />
 			<channel id="motionOffTimer" typeId="motionOffTimer" />

--- a/addons/binding/org.openhab.binding.mihome/ESH-INF/thing/sensor_motion_aq2.xml
+++ b/addons/binding/org.openhab.binding.mihome/ESH-INF/thing/sensor_motion_aq2.xml
@@ -8,7 +8,8 @@
 			<bridge-type-ref id="bridge" />
 		</supported-bridge-type-refs>
 		<label>Xiaomi Aqara Motion Sensor</label>
-		<description>Sensor that detects movement. Also called Occupancy Sensor or Human Body Sensor. After it detects motion, it goes to sleep for 1 minute.</description>
+		<description>Sensor that detects movement. Also called Occupancy Sensor or Human Body Sensor. After it detects motion,
+			it goes to sleep for 1 minute.</description>
 		<channels>
 			<channel id="motion" typeId="motion" />
 			<channel id="illumination" typeId="illumination" />

--- a/addons/binding/org.openhab.binding.mihome/ESH-INF/thing/sensor_switch.xml
+++ b/addons/binding/org.openhab.binding.mihome/ESH-INF/thing/sensor_switch.xml
@@ -8,7 +8,8 @@
 			<bridge-type-ref id="bridge" />
 		</supported-bridge-type-refs>
 		<label>Xiaomi Mi Wireless Switch</label>
-		<description>Round-shaped mini wireless switch that allows to trigger four types of event: single click, double click, long click press, long click release.</description>
+		<description>Round-shaped mini wireless switch that allows to trigger four types of event: single click, double click,
+			long click press, long click release.</description>
 		<channels>
 			<channel id="button" typeId="button" />
 			<channel id="batteryLevel" typeId="system.battery-level" />

--- a/addons/binding/org.openhab.binding.mihome/ESH-INF/thing/sensor_switch_aq2.xml
+++ b/addons/binding/org.openhab.binding.mihome/ESH-INF/thing/sensor_switch_aq2.xml
@@ -8,7 +8,8 @@
 			<bridge-type-ref id="bridge" />
 		</supported-bridge-type-refs>
 		<label>Xiaomi Aqara Wireless Switch</label>
-		<description>Square-shaped mini wireless switch that allows to trigger four types of event: single click, double click, long click press, long click release.</description>
+		<description>Square-shaped mini wireless switch that allows to trigger four types of event: single click, double
+			click, long click press, long click release.</description>
 		<channels>
 			<channel id="button" typeId="button" />
 			<channel id="batteryLevel" typeId="system.battery-level" />

--- a/addons/binding/org.openhab.binding.mihome/README.md
+++ b/addons/binding/org.openhab.binding.mihome/README.md
@@ -100,6 +100,7 @@ Bridge mihome:bridge:f0b429XXXXXX "Xiaomi Gateway" [ serialNumber="f0b429XXXXXX"
 
 ```
 // Xiaomi Gateway
+Switch Gateway_AddDevice { channel="mihome:gateway:<ID>:joinPermission" }
 Switch Gateway_LightSwitch <light> { channel="mihome:gateway:<ID>:brightness" }
 Dimmer Gateway_Brightness <dimmablelight> { channel="mihome:gateway:<ID>:brightness" }
 Color Gateway_Color <rgb> { channel="mihome:gateway:<ID>:color" }

--- a/addons/binding/org.openhab.binding.mihome/src/main/java/org/openhab/binding/mihome/XiaomiGatewayBindingConstants.java
+++ b/addons/binding/org.openhab.binding.mihome/src/main/java/org/openhab/binding/mihome/XiaomiGatewayBindingConstants.java
@@ -77,6 +77,8 @@ public class XiaomiGatewayBindingConstants {
     public static final String CHANNEL_CUBE_ACTION = "action";
     public static final String CHANNEL_CUBE_ROTATION_ANGLE = "rotationAngle";
     public static final String CHANNEL_CUBE_ROTATION_TIME = "rotationTime";
+    // gateway general
+    public static final String CHANNEL_GATEWAY_JOIN_PERMISSION = "joinPermission";
     // gateway sound
     public static final String CHANNEL_GATEWAY_SOUND_SWITCH = "enableSound";
     public static final String CHANNEL_GATEWAY_SOUND = "sound";

--- a/addons/binding/org.openhab.binding.mihome/src/main/java/org/openhab/binding/mihome/handler/XiaomiActorGatewayHandler.java
+++ b/addons/binding/org.openhab.binding.mihome/src/main/java/org/openhab/binding/mihome/handler/XiaomiActorGatewayHandler.java
@@ -10,6 +10,9 @@ package org.openhab.binding.mihome.handler;
 
 import static org.openhab.binding.mihome.XiaomiGatewayBindingConstants.*;
 
+import java.util.Timer;
+import java.util.TimerTask;
+
 import org.eclipse.smarthome.core.library.types.DecimalType;
 import org.eclipse.smarthome.core.library.types.HSBType;
 import org.eclipse.smarthome.core.library.types.OnOffType;
@@ -41,10 +44,17 @@ public class XiaomiActorGatewayHandler extends XiaomiActorBaseHandler {
     private static final String ILLUMINATION = "illumination";
     private static final String MID = "mid";
     private static final String VOL = "vol";
+    private static final String JOIN_PERMISSION = "join_permission";
+    private static final String YES = "yes";
+    private static final String NO = "no";
 
     private Integer lastBrigthness;
     private Integer lastVolume;
     private Integer lastColor;
+
+    private static final int INCLUSION_MODE_TIMEOUT_MILLIS = 30000;
+    private Timer timer;
+    private boolean timerIsActive;
 
     private final Logger logger = LoggerFactory.getLogger(XiaomiActorGatewayHandler.class);
 
@@ -53,11 +63,37 @@ public class XiaomiActorGatewayHandler extends XiaomiActorBaseHandler {
         lastBrigthness = DEFAULT_BRIGTHNESS_PCENT;
         lastVolume = DEFAULT_VOLUME_PCENT;
         lastColor = DEFAULT_COLOR;
+        timerIsActive = false;
+    }
+
+    private class TimerAction extends TimerTask {
+        @Override
+        public synchronized void run() {
+            updateState(CHANNEL_GATEWAY_JOIN_PERMISSION, OnOffType.OFF);
+            timerIsActive = false;
+        }
     }
 
     @Override
     void execute(ChannelUID channelUID, Command command) {
         switch (channelUID.getId()) {
+            case CHANNEL_GATEWAY_JOIN_PERMISSION:
+                if (command instanceof OnOffType) {
+                    if (command == OnOffType.ON) {
+                        timer = new Timer();
+                        timer.schedule(new TimerAction(), INCLUSION_MODE_TIMEOUT_MILLIS);
+                        timerIsActive = true;
+                    } else {
+                        if (timerIsActive) {
+                            timer.cancel();
+                            timerIsActive = false;
+                        }
+                    }
+                    getXiaomiBridgeHandler().writeToBridge(new String[] { JOIN_PERMISSION },
+                            new Object[] { command == OnOffType.ON ? YES : NO });
+                    return;
+                }
+                break;
             case CHANNEL_BRIGHTNESS:
                 if (command instanceof PercentType) {
                     int newBright = ((PercentType) command).intValue();

--- a/addons/binding/org.openhab.binding.mihome/src/main/java/org/openhab/binding/mihome/internal/ModelMapper.java
+++ b/addons/binding/org.openhab.binding.mihome/src/main/java/org/openhab/binding/mihome/internal/ModelMapper.java
@@ -35,7 +35,9 @@ public class ModelMapper {
         THING_MAP.put("ctrl_neutral1", THING_TYPE_ACTOR_AQARA1);
         THING_MAP.put("ctrl_neutral2", THING_TYPE_ACTOR_AQARA2);
         THING_MAP.put("ctrl_ln1", THING_TYPE_ACTOR_AQARA_ZERO1);
+        THING_MAP.put("ctrl_ln1.aq1", THING_TYPE_ACTOR_AQARA_ZERO1);
         THING_MAP.put("ctrl_ln2", THING_TYPE_ACTOR_AQARA_ZERO2);
+        THING_MAP.put("ctrl_ln2.aq1", THING_TYPE_ACTOR_AQARA_ZERO2);
         THING_MAP.put("86sw1", THING_TYPE_SENSOR_AQARA1);
         THING_MAP.put("86sw2", THING_TYPE_SENSOR_AQARA2);
         THING_MAP.put("cube", THING_TYPE_SENSOR_CUBE);
@@ -44,12 +46,12 @@ public class ModelMapper {
         THING_MAP.put("motion", THING_TYPE_SENSOR_MOTION);
         THING_MAP.put("natgas", THING_TYPE_SENSOR_GAS);
         THING_MAP.put("sensor_magnet.aq2", THING_TYPE_SENSOR_AQARA_MAGNET);
-        THING_MAP.put("smoke", THING_TYPE_SENSOR_SMOKE);
+        THING_MAP.put("sensor_motion.aq2", THING_TYPE_SENSOR_AQARA_MOTION);
         THING_MAP.put("sensor_wleak.aq1", THING_TYPE_SENSOR_WATER);
         THING_MAP.put("sensor_switch.aq2", THING_TYPE_SENSOR_AQARA_SWITCH);
+        THING_MAP.put("smoke", THING_TYPE_SENSOR_SMOKE);
         THING_MAP.put("switch", THING_TYPE_SENSOR_SWITCH);
         THING_MAP.put("weather.v1", THING_TYPE_SENSOR_AQARA_WEATHER_V1);
-        THING_MAP.put("sensor_motion.aq2", THING_TYPE_SENSOR_AQARA_MOTION);
 
         LABEL_MAP.put("curtain", "Xiaomi Aqara Intelligent Curtain Motor");
         LABEL_MAP.put("gateway", "Xiaomi Mi Smart Home Gateway");
@@ -57,7 +59,9 @@ public class ModelMapper {
         LABEL_MAP.put("ctrl_neutral1", "Xiaomi Aqara Wall Switch 1 Button");
         LABEL_MAP.put("ctrl_neutral2", "Xiaomi Aqara Wall Switch 2 Button");
         LABEL_MAP.put("ctrl_ln1", "Xiaomi \"zero-fire\" 1 Channel Wall Switch");
+        LABEL_MAP.put("ctrl_ln1.aq1", "Xiaomi Aqara \"zero-fire\" 1 Channel Wall Switch");
         LABEL_MAP.put("ctrl_ln2", "Xiaomi \"zero-fire\" 2 Channel Wall Switch");
+        LABEL_MAP.put("ctrl_ln2.aq1", "Xiaomi \"zero-fire\" 2 Channel Wall Switch");
         LABEL_MAP.put("86sw1", "Xiaomi Aqara Smart Switch 1 Button");
         LABEL_MAP.put("86sw2", "Xiaomi Aqara Smart Switch 2 Button");
         LABEL_MAP.put("cube", "Xiaomi Mi Smart Cube");
@@ -66,12 +70,12 @@ public class ModelMapper {
         LABEL_MAP.put("motion", "Xiaomi Mi Motion Sensor");
         LABEL_MAP.put("natgas", "Xiaomi Mijia Honeywell Gas Alarm Detector");
         LABEL_MAP.put("sensor_magnet.aq2", "Xiaomi Aqara Door/Window Sensor");
-        LABEL_MAP.put("water", "Xiaomi Aqara Water Leak Sensor");
-        LABEL_MAP.put("smoke", "Xiaomi Mijia Honeywell Fire Alarm Detector");
+        LABEL_MAP.put("sensor_motion.aq2", "Xiaomi Aqara Motion Sensor");
+        LABEL_MAP.put("sensor_wleak.aq1", "Xiaomi Aqara Water Leak Sensor");
         LABEL_MAP.put("sensor_switch.aq2", "Xiaomi Aqara Wireless Switch");
+        LABEL_MAP.put("smoke", "Xiaomi Mijia Honeywell Fire Alarm Detector");
         LABEL_MAP.put("switch", "Xiaomi Mi Wireless Switch");
         LABEL_MAP.put("weather.v1", "Xiaomi Aqara Temperature, Humidity & Pressure Sensor");
-        LABEL_MAP.put("sensor_motion.aq2", "Xiaomi Aqara Motion Sensor");
     }
 
     public static ThingTypeUID getThingTypeForModel(String model) {

